cns/service_test.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package cns
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-container-networking/cns/common"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	acn "github.com/Azure/azure-container-networking/common"
+	serverTLS "github.com/Azure/azure-container-networking/server/tls"
+	"github.com/Azure/azure-container-networking/store"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewService(t *testing.T) {
```

Comment:
It will be good to add a test case for TLSPort == "" also.

Reply:
added a check for `config.TLSSettings` being empty in the non-TLS case